service/history/workflow_task_handler.go

```diff
@@ -345,6 +349,15 @@ func (handler *workflowTaskHandlerImpl) handleCommand(
 		}
 		return nil, err
 	}
+
+	if err != nil {
+		return nil, err
+	}
+
+	if historyEvent != nil && command.UserMetadata != nil {
+		historyEvent.UserMetadata = command.UserMetadata
```

Comment:
I get the idea that UserMetadata applies to all history events so it's a field on the Command type not on individual command attributes. But my understanding is ^ is not the case for history events?

Reply:
Synced offline. This is to keep the server changes minimum.